### PR TITLE
refactor: replace &PathBuf with &Path to enhance generality

### PR DESCRIPTION
**Pull request by Integral-Tech**
- According to [style.md](https://github.com/rust-lang/rust/blob/master/src/tools/rust-analyzer/docs/dev/style.md#useless-types):

> More generally, always prefer types on the left
```rust
// GOOD      BAD
&[T]         &Vec<T>
&str         &String
Option<&T>   &Option<T>
&Path        &PathBuf
```

### CODE REVIEW

**Review by compiler-errors — CHANGES_REQUESTED** [review]
Hi. Thanks for the contribution, but please revert changes to subtrees like rustfmt.

**Comment by Integral-Tech** [comment]
@compiler-errors Done :)

**Comment by compiler-errors** [comment]
Generally I want to discourage using the *rust-analyzer* style guide to make basically non-functional changes to rust-lang/rust, but since this is already done I'll approve it.

@bors r+ rollup

File: src/tools/rustc-perf-wrapper/src/main.rs [hunk 1]
```diff
@@ -163,7 +163,7 @@ fn apply_shared_opts(cmd: &mut Command, opts: &SharedOpts) {
     }
 }
 
-fn execute_benchmark(cmd: &mut Command, compiler: &PathBuf) {
+fn execute_benchmark(cmd: &mut Command, compiler: &Path) {
```
Comment (safinaskar) [inline src/tools/rustc-perf-wrapper/src/main.rs:166]: @Integral-Tech , @compiler-errors , now I see error `cannot find type `Path` in this scope`, when I try to run `./x perf -- benchmark id`. Why this file doesn't get compiled in CI?

Comment (Integral-Tech) [reply src/tools/rustc-perf-wrapper/src/main.rs:166]: @safinaskar Fixed in another PR #134816

**Pull request merged**